sphinx/io.py

```diff
@@ -53,6 +53,8 @@ def __init__(self, app, parsers={}, *args, **kwargs):
             parser = parser_class()
             if hasattr(parser, 'set_application'):
                 parser.set_application(app)
+            if hasattr(parser, 'set_smartquotes_allowed'):
+                parser.set_smartquotes_allowed(app.env.settings['smart_quotes'])
```

Comment:
as far as I understand, this will be called from a BuildEnvironment object ``read_doc()`` method, when a ``reader`` is being set up, and at that time the ``app.env.settings['smart_quotes']`` always exists with value either True or False due to earlier code executed by the BuildEnvironment instance.

Now, perhaps the accessor method is superfluous here and we should set the Boolean directly rather than having such an instance method added to RSTParser class.